File: tlg_nodejs/tlg/HighscoreEntry.ts
export class HighscoreEntry {

        private static readonly NAMEBUFFER_LIMIT:number=20;
        private static readonly DEFAULT_NAME:string="-";

        public readonly name:string;
        public readonly score:number;

        constructor(n:string, s:number) {
                this.name = n;
                this.score = s;
        }
}



        /*
        public HighscoreEntry(BufferedInputStream reader) throws IOException {
                score = ByteInteger.read(reader);
                name = readName(reader);
        }


        private String readName(BufferedInputStream reader) throws IOException {
                byte[] buf;
                int len = ByteInteger.read(reader);

                if (len > NAMEBUFFER_LIMIT || len < 1) throw new IOException();
                buf= new byte[len];
                reader.read(buf);
                return new String(buf);
        }



        public void writeState(BufferedOutputStream output) throws IOException {
        System.err.print("write entry");
        System.err.print(name + "\n");


                byte[] buf=name.getBytes();
                ByteInteger.wrap(score).writeState(output);
                ByteInteger.wrap(buf.length).writeState(output);
                output.write(buf);
        }

}*/

File: tlg/src/main/kotlin/score/HighScoreEntry.kt
package score

import context.ByteInteger
import java.io.BufferedInputStream
import java.io.BufferedOutputStream
import java.io.IOException


class HighscoreEntry {
    val name: String
    val score: Int

    constructor() {
        score = 0
        name = DEFAULT_NAME
    }

    constructor(n: String, s: Int) {
        name = n
        score = s
    }

    constructor(reader: BufferedInputStream) {
        score = ByteInteger.read(reader)
        name = readName(reader)
    }

    @Throws(IOException::class)
    private fun readName(reader: BufferedInputStream): String {
        val buf: ByteArray
        val len: Int = ByteInteger.read(reader)
        if (len > NAMEBUFFER_LIMIT || len < 1) throw IOException()
        buf = ByteArray(len)
        reader.read(buf)
        return String(buf)
    }

    @Throws(IOException::class)
    fun writeState(output: BufferedOutputStream) {
        System.err.print("write entry")
        System.err.print(
            """
                $name
                
                """.trimIndent()
        )
        val buf = name.toByteArray()
        ByteInteger.wrap(score).writeState(output)
        ByteInteger.wrap(buf.size).writeState(output)
        output.write(buf)
    }

    companion object {
        private const val NAMEBUFFER_LIMIT = 20
        private const val DEFAULT_NAME = "-"
    }
}
